Common/ImageSamplers/itkImageGridSampler.hxx

```diff
@@ -255,7 +257,9 @@ ImageGridSampler<TInputImage>::SingleThreadedGenerateData(const TInputImage &
             // Translate index to point.
             inputImage.TransformIndexToPhysicalPoint(index, tempSample.m_ImageCoordinates);
 
-            if (mask->IsInsideInWorldSpace(tempSample.m_ImageCoordinates))
+            // Equivalent to `mask->IsInsideInWorldSpace(tempSample.m_ImageCoordinates)`, but much faster.
+            if (mask->IsInsideInObjectSpace(
+                  worldToObjectTransform.WorldToObjectTransformType::TransformPoint(tempSample.m_ImageCoordinates)))
```

Comment:
@mstaring Thanks for looking at the issue! The original elastix `itk::ImageGridSampler` (before this commit) called `SpatialObject::IsInsideInWorldSpace`, which is implemented by ITK, [as follows](https://github.com/InsightSoftwareConsortium/ITK/blob/0b47b7bf95665eb31e00442074a3a77e11e1f790/Modules/Core/SpatialObjects/include/itkSpatialObject.hxx#L187-L194):

```
SpatialObject<TDimension>::IsInsideInWorldSpace(const PointType &   point,
                                                unsigned int        depth,
                                                const std::string & name) const
{
  PointType pnt;
  pnt = this->GetObjectToWorldTransformInverse()->TransformPoint(point);
  return IsInsideInObjectSpace(pnt, depth, name);
}

```

`this->GetObjectToWorldTransformInverse()` is a rather expensive call, so this PR takes that call out of the nested for loops. Also it saves time by skipping the `depth` and `name` arguments. And then it bypasses the virtual table, by doing a non-virtual call to `TransformPoint` (because the runtime-time of worldToObjectTransform is already known at compile-time).

But maybe I misunderstand your question...?  🤷